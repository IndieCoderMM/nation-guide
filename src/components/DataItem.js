import React from 'react';

const DataItem = ({ title, data, unit = '' }) => {
  return (
    <div className="data-item">
      <h3 className="data-title">{title}</h3>
      <div className="data-value">
        <p>{data}</p>
        <p>{unit}</p>
      </div>
    </div>
  );
};

export default DataItem;
