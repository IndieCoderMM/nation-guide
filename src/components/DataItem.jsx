import PropTypes from 'prop-types';
import styles from './DataItem.module.css';

const DataItem = ({ title, data, unit }) => (
  <div className={styles.container}>
    <h4 className={styles.title}>{title}</h4>
    <div className={styles.data}>
      <p>
        {data}&nbsp;{unit}
      </p>
    </div>
  </div>
);

DataItem.propTypes = {
  title: PropTypes.string,
  data: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  unit: PropTypes.string,
};

DataItem.defaultProps = {
  title: '',
  data: 0,
  unit: '',
};

export default DataItem;
