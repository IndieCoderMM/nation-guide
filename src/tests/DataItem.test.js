import React from 'react';
import renderer from 'react-test-renderer';
import DataItem from '../components/DataItem';

describe('Component for Displaying Data', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<DataItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
