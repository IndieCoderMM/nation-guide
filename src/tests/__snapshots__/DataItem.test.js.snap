// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component for Displaying Data should render correctly 1`] = `
<DocumentFragment>
  <div
    class="data-item"
  >
    <h3
      class="data-title"
    >
      Area
    </h3>
    <div
      class="data-value"
    >
      <p>
        9984670
      </p>
      <p />
    </div>
  </div>
</DocumentFragment>
`;
